examples/js/exporters/GLTFExporter.js

```diff
@@ -707,16 +708,27 @@ THREE.GLTFExporter.prototype = {
 
 				var canvas = cachedCanvas = cachedCanvas || document.createElement( 'canvas' );
 
-				canvas.width = image.width;
-				canvas.height = image.height;
+				canvas.width = Math.min( image.width, options.maxTextureSize );
+				canvas.height = Math.min( image.height, options.maxTextureSize );
 
-				if ( options.forcePowerOfTwoTextures && ! isPowerOfTwo( image ) ) {
+				if ( options.forcePowerOfTwoTextures) {
 
-					console.warn( 'GLTFExporter: Resized non-power-of-two image.', image );
+					if(! isPowerOfTwo( image )) {
+						console.warn( 'GLTFExporter: Resized non-power-of-two image.', image );
+					}
 
-					canvas.width = THREE.Math.floorPowerOfTwo( canvas.width );
-					canvas.height = THREE.Math.floorPowerOfTwo( canvas.height );
+					if(! THREE.Math.isPowerOfTwo( canvas.width )) {
+						console.warn( 'GLTFExporter: power-of-two textures requested, but the given maxTextureSize is not power-of-two.')
+					}
 
+					// REMARK: These 'if' checks are performed just to avoid the unnecessary POW+LOG calls behind floorPowerOfTwo
+					if(! THREE.Math.isPowerOfTwo( canvas.width )) {
+						canvas.width = THREE.Math.floorPowerOfTwo( canvas.width );
+					}
+					
+					if(! THREE.Math.isPowerOfTwo( canvas.height )) {
+						canvas.height = THREE.Math.floorPowerOfTwo( canvas.height );
+					}
```

Comment:
After the suggestion above, these two if blocks can be removed.

Reply:
Just this would be cleaner, without the if blocks:

```js
canvas.width = THREE.Math.floorPowerOfTwo( canvas.width );
canvas.height = THREE.Math.floorPowerOfTwo( canvas.height );
```

The cost of computing a POT is trivial compared to writing out the data.